packages/admin/cms-admin/src/generator/future/generateGrid/combinationColumn.ts

```diff
@@ -64,21 +64,27 @@ type CellContent = {
 };
 
 const getTextForCellContent = (textConfig: TextConfig<string>, messageIdPrefix: string, target: "primary" | "secondary"): CellContent => {
-    if (typeof textConfig === "string") {
+    if (typeof textConfig !== "string" && textConfig.type === "static") {
         return {
-            textContent: `row.${textConfig}`,
+            textContent: getFormattedMessageNode(messageIdPrefix, textConfig.text),
         };
     }
 
-    if (textConfig.type === "static") {
+    const emptyText =
+        typeof textConfig !== "string" && "emptyValue" in textConfig
```

Comment:
I believe you could remove the typeof checks here 🤔

Reply:
Not really, `textConfig` can be a `string` here, depending on how it was defined in the config.

Reply:
I thought there already was a type check somewhere above, but probably read the wrong line.